web/src/main/java/org/cbioportal/web/ClinicalDataController.java

```diff
@@ -38,7 +38,7 @@
     @RequestMapping(value = "/studies/{studyId}/samples/{sampleId}/clinical-data", method = RequestMethod.GET,
             produces = MediaType.APPLICATION_JSON_VALUE)
     @ApiOperation("Get all clinical data of a sample in a study")
-    public ResponseEntity<List<? extends ClinicalDataSummary>> getAllClinicalDataOfSampleInStudy(
+    public ResponseEntity<List<ClinicalData>> getAllClinicalDataOfSampleInStudy(
             @ApiParam(required = true, value = "Study ID e.g. acc_tcga")
```

Comment:
I agree that we need to put more effort to make service and persistence layers foolproof, I'm currently more focused on Web API level so there are some parts needs to be improved. We can revisit these parts later when we get feedback from 3rd party users of those layers.